src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -545,6 +545,54 @@ async Task<ArraySegment<byte>> WriteLiteralExpressionAsIndex(DotnetObjectId root
             }
         }
 
+        private static bool CheckParametersCompatibility(ElementType? typeCode, JObject value)
+        {
+            if (!typeCode.HasValue)
+                return true;
+            var parameterType = value["type"]?.Value<string>();
+            var parameterClassName = value["className"]?.Value<string>();
+
+            switch (typeCode.Value)
+            {
+                case ElementType.Object:
+                    if (parameterType != "object")
+                        return false;
```

Comment:
Do IIUC that we are checking if value can be passed as an argument to a method with parameter type typeCode?
If so, then shouldn't param type == `object` accept anything?

Reply:
If a method is expecting an object, can we call it passing a integer? I don't think so.